experiments/ssl4eo/sample_conus.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import csv
+import os
+import random
+
+import fiona
+from rtree import index
+from sample_ssl4eo import create_bbox, km2deg
+from shapely.geometry import MultiPolygon, Point, shape
+from shapely.ops import unary_union
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+
+def retrieve_rois_polygons(download_root: str) -> MultiPolygon:
+    nation_url = (
+        "https://www2.census.gov/geo/tiger/GENZ2022/shp/cb_2022_us_nation_5m.zip"
+    )
+    nation_filename = "cb_2022_us_nation_5m.shp"
+    download_and_extract_archive(nation_url, download_root)
+
+    state_url = "https://www2.census.gov/geo/tiger/GENZ2018/shp/cb_2018_us_state_5m.zip"
+    state_filename = "cb_2018_us_state_5m.shp"
+    download_and_extract_archive(state_url, download_root)
+
+    excluded_states = [
+        "United States Virgin Islands",
+        "Commonwealth of the Northern Mariana Island",
+        "Puerto Rico",
+        "Alaska",
+        "Hawaii",
+        "American Samoa",
+        "Guam",
+    ]
+    with fiona.open(os.path.join(download_root, state_filename), "r") as shapefile:
+        excluded = []
+        for feature in shapefile:
+            name = feature["properties"]["NAME"]
+            if name in excluded_states:
+                excluded.append(shape(feature["geometry"]))
+
+    with fiona.open(os.path.join(download_root, nation_filename), "r") as shapefile:
+        usa = shape(shapefile[0]["geometry"])  # still usa
+
+    excluded_polygons = unary_union(excluded)
+    conus = usa.difference(excluded_polygons)
```

Comment:
Instead of computing the union of the exclusions and removing them from USA, could we instead compute the union of CONUS and not even need the USA bounds?